### PR TITLE
General user statistics improvements #33

### PR DESCRIPTION
**Pull request by rosinbum**
This is a general improvement ticket for the User Statistics page.

1) Order the data table so that the most recent date is first
2) Move graph to the top of the page because that is the quick and easy way to view the data.
3) Add another graph to see the followers/following numbers
4) Add labels to the graphs
5) Fix negative new followers graph bug

Screen shot of proposed changes:
Screen Shot 2019-04-26 at 2 44 16 PM

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -14433,6 +14433,11 @@
       "resolved": "https://registry.npmjs.org/react-lifecycles-compat/-/react-lifecycles-compat-3.0.4.tgz",
       "integrity": "sha512-fBASbA6LnOU9dOU2eW7aQ8xmYBSXUIWr+UmF9b1efZBazGNO+rcXT/icdKnYm2pTwcRylVUYwW7H1PHfLekVzA=="
     },
+    "react-moment": {
+      "version": "0.9.2",
+      "resolved": "https://registry.npmjs.org/react-moment/-/react-moment-0.9.2.tgz",
+      "integrity": "sha512-jpKIEvcEOTSa4RXWTjDlvja7sN+ee5Gyk5ZKa704FZxE01OxmKIpq9GUliqTrAgdDYJJUQsnqQamczn0TN+DTg=="
+    },
```
Comment (rosinbum) [inline package-lock.json]: Using moment for date compare

File: src/components/UserStatistics/index.jsx [hunk 2]
```diff
@@ -69,7 +70,7 @@ class AccountStatistics extends Component {
             if (newFollowers > 0) {
               newFollowers = `+${newFollowers}`
             } else if (newFollowers < 0) {
-              newFollowers = `-${newFollowers}`
+              newFollowers = `${newFollowers}`
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: The number is already negative. In the data table it shows up with two minus signs `--` and that brakes the graph. So just remove it to makes things work correctly

File: src/components/UserStatistics/index.jsx [hunk 3]
```diff
@@ -105,28 +106,42 @@ class AccountStatistics extends Component {
     if (this.state.updatedUserStats.length !== 0) {
       rows = this.state.updatedUserStats
     }
+
+    const compareRowsByDateDesc =(a, b) => {
+      return Moment(b.day).diff(Moment(a.day))
+    }
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: New compare function to sort the data table by date descending

File: src/components/UserStatistics/index.jsx [hunk 4]
```diff
@@ -105,28 +106,42 @@ class AccountStatistics extends Component {
...
-    let chartData = {}
-    rows.map((item) =>
-      chartData[item.day] = item.newFollowers
+    let followersFollowingChartData = [{name: 'Followers', data: {}}, {name: 'Following', data: {}}]
+    let newFollowersChartData = [{name: 'New Followers', data: {}}]
+    rows.forEach((item) => {
+        followersFollowingChartData[0]['data'][item.day] = item.followers
+        followersFollowingChartData[1]['data'][item.day] = item.following
+        newFollowersChartData[0]['data'][item.day] = item.newFollowers
+      }
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: Change here to add data to both the new followers chart data and the followers/following chart

File: src/components/UserStatistics/index.jsx [hunk 5]
```diff
@@ -105,28 +106,42 @@ class AccountStatistics extends Component {
...
     )
+
     return (
       <div className={classes.wrapper}>
         <h1>User Statistics</h1>
+        <Paper>
+          <LineChart data={followersFollowingChartData} />
+          <LineChart data={newFollowersChartData} />
+        </Paper>
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: Moved the charts to the top of the page and added the new following/followers chart

File: src/components/UserStatistics/index.jsx [hunk 6]
```diff
@@ -105,28 +106,42 @@ class AccountStatistics extends Component {
...
           <Table className={classes.table}>
             <TableHead>
               <TableRow>
                 <TableCell>New Followers</TableCell>
-                <TableCell>followers</TableCell>
-                <TableCell>following</TableCell>
-                <TableCell>created</TableCell>
+                <TableCell>Followers</TableCell>
+                <TableCell>Following</TableCell>
+                <TableCell>Created</TableCell>
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: Capitalizing the data table headers to match `New Following`

File: src/components/UserStatistics/index.jsx [hunk 7]
```diff
@@ -105,28 +106,42 @@ class AccountStatistics extends Component {
...
               </TableRow>
             </TableHead>
             <TableBody>
-              {rows.map(row => {
+              {rows.sort(compareRowsByDateDesc).map(row => {
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: Sort the data table by date descending before rendering

File: src/components/UserStatistics/index.jsx [hunk 8]
```diff
@@ -145,9 +160,7 @@ class AccountStatistics extends Component {
           />
 
         </Paper>
-        <Paper>
-          <LineChart data={chartData} />
-        </Paper>
+
```
Comment (rosinbum) [inline src/components/UserStatistics/index.jsx]: moved this chart to the top. Line 129

**Comment by converge** [comment]
thanks @rosinbum for your contribution, great work, and good ideas. 🍻

**Comment by rosinbum** [comment]
@converge no problem

**Pull request merged**